extended/src/main/java/apoc/vectordb/VectorDb.java

```diff
@@ -0,0 +1,254 @@
+        }
+        if (MapUtils.isEmpty(metadata)) {
+            throw new RuntimeException("To use mapping config, the metadata should not be empty. Make sure you execute `YIELD metadata` on the procedure");
+        }
+        Map<String, Object> metaProps = new HashMap<>(metadata);
+        if (mapping.getLabel() != null) {
+            return handleMappingNode(tx, mapping, metaProps, embedding);
+        } else if (mapping.getType() != null) {
+            return handleMappingRel(tx, mapping, metaProps, embedding);
+        } else {
+            throw new RuntimeException("Mapping conf has to contain either label or type key");
+        }
+    }
+
+    private static Entity handleMappingNode(Transaction transaction, VectorMappingConfig mapping, Map<String, Object> metaProps, List<Double> embedding) {
+        try {
+            Node node;
+            Object propValue = metaProps.get(mapping.getId());
+            node = transaction.findNode(Label.label(mapping.getLabel()), mapping.getProp(), propValue);
+            if (node == null && mapping.isCreate()) {
+                node = transaction.createNode(Label.label(mapping.getLabel()));
+                node.setProperty(mapping.getProp(), propValue);
+            }
+            if (node != null) {
+                setProperties(node, metaProps);
+                setVectorProp(mapping, embedding, node);
+            }
+
+            return node;
+        } catch (MultipleFoundException e) {
+            throw new RuntimeException("Multiple nodes found");
+        }
+    }
+
+    private static Entity handleMappingRel(Transaction transaction, VectorMappingConfig mapping, Map<String, Object> metaProps, List<Double> embedding) {
+        try {
+            // in this case we cannot auto-create the rel, since we should have to define start and end node as well
+            Relationship rel;
+            Object propValue = metaProps.get(mapping.getId());
+            rel = transaction.findRelationship(RelationshipType.withName(mapping.getType()), mapping.getProp(), propValue);
+            if (rel != null) {
+                setProperties(rel, metaProps);
+                setVectorProp(mapping, embedding, rel);
+            }
+
+            return rel;
+        } catch (MultipleFoundException e) {
+            throw new RuntimeException("Multiple relationships found");
+        }
+    }
+
+    private static <T extends Entity> void setVectorProp(VectorMappingConfig mapping, List<Double> embedding, T entity) {
+        if (mapping.getEmbeddingProp() == null) {
+            return;
+        }
+
+        if (embedding == null) {
+            String embeddingErrMsg = "The embedding value is null. Make sure you execute `YIELD embedding` on the procedure and you configured `%s: true`"
+                    .formatted(ALL_RESULTS_KEY);
+            throw new RuntimeException(embeddingErrMsg);
+        }
+
+        entity.setProperty(mapping.getEmbeddingProp(), embedding.stream()
```

Comment:
I think we should just do a utility method that creates a float array of the list size and uses a for loop over the list to set the values. then the JVM can also optimize that to SIMD. I don't think that the streams are efficient here.